### PR TITLE
move require SSL to it's own doc

### LINKED ISSUE #135: Enforcing SSL
ASP.NET MVC -> Security -> Enforcing SSL

### PR DESCRIPTION
**Pull request by Rick-Anderson**
fixes #135

### CODE REVIEW

File: aspnetcore/security/enforcing-ssl.md [hunk 1]
```diff
@@ -1,18 +1,46 @@
...
 uid: security/enforcing-ssl
 ---
-# 🔧 Enforcing SSL
+# Enforcing SSL in an ASP.NET Core app
 
-> [!NOTE]
-> We are currently working on this topic.We welcome your input to help shape the scope and approach. You can track the status and provide input on this [issue](https://github.com/aspnet/Docs/issues/135) at GitHub.If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the [issue](https://github.com/aspnet/Docs/issues/135).Learn more about how you can [contribute](https://github.com/aspnet/Docs/blob/master/CONTRIBUTING.md) on GitHub.
+This document shows how to:
+
+- Require all requests use SSL (that is are HTTPS requests).
```
Comment (tdykstra) [inline aspnetcore/security/enforcing-ssl.md]: Require SSL for all requests

File: aspnetcore/security/enforcing-ssl.md [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
+- Redirect all HTTP requests to HTTPS.
+- Set up IIS Express to use SSL/HTTPS.
+
+## Require SSL
+
+The [RequireHttpsAttribute](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.requirehttpsattribute) is used to require request use HTTPS. You can decorate controllers or methods with this attribute or you can apply it globally as shown below:
```
Comment (tdykstra) [inline aspnetcore/security/enforcing-ssl.md]: is used to require SSL

File: aspnetcore/security/enforcing-ssl.md [hunk 3]
```diff
@@ -1,18 +1,46 @@
...
+
+* Add the following code to `ConfigureServices` in `Startup`:
+
+[!code-csharp[Main](authentication/accconfirm/sample/WebApp1/Startup.cs?name=snippet2&highlight=4-)]
```
Comment (tdykstra) [inline aspnetcore/security/enforcing-ssl.md]: Using snippet code from a different article's samples folder seems likely to cause maintenance issues when someone changes the sample and doesn't realize it's used elsewhere.  Consider adding comments in the sample project warning that the code is referenced elsewhere.  Consider using the Common folder which already has a project that might need minor updates for this article.

File: aspnetcore/security/enforcing-ssl.md [hunk 4]
```diff
@@ -1,18 +1,46 @@
...
+
+- Require all requests use SSL (that is are HTTPS requests).
+- Redirect all HTTP requests to HTTPS.
+- Set up IIS Express to use SSL/HTTPS.
+
+## Require SSL
+
+The [RequireHttpsAttribute](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.requirehttpsattribute) is used to require request use HTTPS. You can decorate controllers or methods with this attribute or you can apply it globally as shown below:
+
+* Add the following code to `ConfigureServices` in `Startup`:
```
Comment (tdykstra) [inline aspnetcore/security/enforcing-ssl.md]: Why is this bulleted?

File: aspnetcore/security/enforcing-ssl.md [hunk 5]
```diff
@@ -1,18 +1,46 @@
...
+
+[!code-csharp[Main](authentication/accconfirm/sample/WebApp1/Startup.cs?name=snippet2&highlight=4-)]
+
+The highlighted code above requires all requests to the app use `HTTPS`. HTTP requests will be ignored. The following highlighted code redirects all HTTP requests to HTTPS:
```
Comment (tdykstra) [inline aspnetcore/security/enforcing-ssl.md]: ~~The highlighted code above requires all requests to the app use `HTTPS`. HTTP requests will be ignored.~~ The highlighted code above causes HTTP requests to be ignored.  The following highlighted code redirects all HTTP requests to HTTPS:

File: aspnetcore/security/enforcing-ssl.md [hunk 6]
```diff
@@ -1,18 +1,46 @@
...
+
+[!code-csharp[Main](authentication/accconfirm/sample/WebApp1/Startup.cs?name=snippet_AddRedirectToHttps&highlight=6-)]
+
+See [URL Rewriting Middleware](xref:fundamentals/url-rewriting) for more information.
+
+Requiring HTTPS globally (`options.Filters.Add(new RequireHttpsAttribute());`) is considered a security best practice. Applying the `[RequireHttps]` to controllers has the drawback that you're not guaranteed new controllers added to you project will get this protection.
```
Comment (tdykstra) [inline aspnetcore/security/enforcing-ssl.md]: is ~~considered~~ a security best practice.

~~you're not guaranteed~~ new controllers might get added to your project without this protection.

File: aspnetcore/security/enforcing-ssl.md [hunk 7]
```diff
@@ -1,18 +1,46 @@
...
+
+## Set up IIS Express for SSL/HTTPS
+
+   * In Solution Explorer, right click the project and select **Properties**.
+   * On the left pane, select **Debug**.
+   * Check **Enable SSL**
+   * Copy the SSL URL and paste it into the **App URL**
+
+![Debug tab of web application properties](enforcing-ssl/_static/ssl.png)
```
Comment (tdykstra) [inline aspnetcore/security/enforcing-ssl.md]: On a new Core web app I follow these instructions and get "An error occurred attempting to determine the process id of dotnet.exe which is hosting your application."

File: aspnetcore/security/enforcing-ssl.md (hunk 3 above)
Comment (Rick-Anderson) [inline aspnetcore/security/enforcing-ssl.md]: // The snippets in here are used by aspnetcore/security/enforcing-ssl.md
        // Any changes to this file should be checked against that doc.

File: aspnetcore/security/enforcing-ssl.md (hunk 6 above)
Comment (Rick-Anderson) [inline aspnetcore/security/enforcing-ssl.md]: reworded to
The highlighted code above requires all requests use `HTTPS`, therefore HTTP requests are ignored. The following highlighted code redirects all HTTP requests to HTTPS:

File: aspnetcore/security/enforcing-ssl.md (hunk 7 above)
Comment (Rick-Anderson) [inline aspnetcore/security/enforcing-ssl.md]: I've seen that error before, not when I was adding SSL. Can you repro this? I can't. Try exit/restart VS, I think that cleared it up for me.

**Pull request merged**